client/packages/react/src/InstantReactAbstractDatabase.tsx

```diff
@@ -345,4 +369,64 @@ export default abstract class InstantReactAbstractDatabase<
   }> => {
     return this._core.queryOnce(query, opts);
   };
+
+  /**
+   * Only render children if the user is signed in.
+   * Optional `loading` prop will render if the user is loading.
+   *
+   * @example
+   *  <db.SignedIn>
+   *    <MyComponent />
+   *  </db.SignedIn>
+   *
+   *  <db.SignedIn loading={<div>Loading...</div>}>
+   *    <MyComponent />
+   *  </db.SignedIn>
+   */
+  SignedIn: React.FC<{
+    children: ReactNode;
+    loading?: ReactNode;
+    error?: ReactNode;
+  }> = ({ children, loading, error }) => {
+    const auth = this.useAuth();
+    if (auth.user) {
```

Comment:
Nit: I would re-write it like this: 

```
`if (isLoading) ...` 
`if (error) ...` 
`if (!user) return null` 
return <>{children}</> 
```

The idea being that all the previous states are a kind of `guard` for this main state.